site/docs/v1/tech/themes/tailwind.adoc

```diff
@@ -0,0 +1,208 @@
+@tailwind components;
+@tailwind utilities;
+```
+
+This stylesheet will be used to import the Tailwind CSS base, components, and utilities. Additionally, you can add custom classes to this stylesheet as well.
+
+Currently, the theme is using the `fusionauth-style.css` hosted in the FusionAuth instance. This could conflict with our Tailwind CSS file. To fix this, we need to remove the following line from the `src/helpers.ftl` file - it should be around line number 85:
+```html
+  <link rel="stylesheet" href="/css/fusionauth-style.css?version=${version}"/>
+```
+
+Finally, we add a new script to the `package.json` file:
+```json
+"scripts": {
+  "watch": "node --dns-result-order ipv4first ./startwatch.js",
+  "watch:tailwind": "npx tailwindcss -i ./input.css -o ./src/styles.css --watch"
+},
+```
+
+This script will build the Tailwind CSS file and output it to the `src` directory.
+
+At this point, we have prepared the project to build the Tailwind CSS file and upload the theme to FusionAuth. To apply the changes in the `helper.ftl` file, run the following command:
+```bash
+./upload.sh
+```
+
+== Usage
+
+There are now two scripts that can be used to build the Tailwind CSS file and upload the theme to FusionAuth:
+
+* `npm run watch` — watch the template directory for changes and upload the theme to FusionAuth
+* `npm run watch:tailwind` — watch the template directory for changes and rebuild the Tailwind CSS file
+
+If you start both scripts in separate terminal windows, you can edit the template files, which rebuild the Tailwind CSS file, and automatically upload the theme to FusionAuth.
+
+Now if we use a Tailwind CSS class in a template, it will be included in the CSS file:
+```html
+<div class="bg-gray-200">
+  <h1 class="text-2xl">Hello World</h1>
+</div>
+```
+
+To preview the changes, open FusionAuth, navigate to [breadcrumb]#Customization -> Themes#. Choose your theme, then click [uielement]#Preview# (the green magnifying glass).
+
+== Integrate DaisyUI
+
+There are many Tailwind CSS UI component libraries available. In this example, we are going to use DaisyUI.
+
+DaisyUI is a Tailwind CSS plugin that provides a set of pre-built components and utilities to build beautiful and responsive websites. It supports color theming, light and dark mode, and RTL.
+
+To integrate DaisyUI, install the plugin:
+```bash
+npm install daisyui
+```
+
+Then add the plugin to the `tailwind.config.js`:
+```js
+module.exports = {
+  content: ['./src/*.ftl'],
+  theme: {
+    extend: {},
+  },
+  plugins: [
+    require('daisyui'),
+  ],
+}
+```
+
+Now you can use the DaisyUI components in the FreeMarker template files:
+```html
+<div class="bg-gray-200">
+  <h1 class="text-2xl">Hello World</h1>
+  <div class="p-4">
+    <button class="btn btn-primary">Primary</button>
+  </div>
+</div>
+```
+
+For an example integration, see link:https://github.com/sonderformat-llc/fusionauth-example-tailwind-daisyui/[FusionAuth + Tailwind + DaisyUI Repository on GitHub].
```

Comment:
Here's the new repository: https://github.com/FusionAuth/fusionauth-example-theme-tailwind-daisyui

Reply:
Can you augment https://github.com/FusionAuth/fusionauth-theme-helper to work with tailwind instead? Or what do you think is a good idea for not having the two get out of sync?

Reply:
Can you please take a screenshot of the login page with this theme applied and add it to the repository?

Reply:
Instead of creating a separate repo, it could be just a fork. Any changes in `fusionauth-theme-helper` could be synced into the fork with GitHub.

As an alternative, the theme helper scripts could be a npm package which can be installed along side tailwind:
```
npm i @fusionauth/scripts
npx fusionauth-theme download $API_KEY $THEME_ID src
npx fusionauth-theme upload $API_KEY $THEME_ID src
```
So these scripts are not part of the users theme repo.

Reply:
@ColinFrick how much time would you estimate it would take to bundle the theme helper scripts into an npm package?

Reply:
I will look into it on Monday. Shouldn't take too much time.

Reply:
I have an npm package prepared: https://github.com/sonderformat-llc/fusionauth-cli

To test it, clone the repository and execute:
```
npm run build
npm install -g .
```

This installs a `fusionauth` command to your cli. This includes - currently - only the theme helper commands:
- `theme:download`
- `theme:upload`
- `theme:watch`

Reply:
@mooreds Is this something I should pursue? Should I go ahead and rewrite the tutorial based on `@fusionauth/cli`?

Reply:
@mooreds great! If you have any functionality you want to see in the cli, besides the theme stuff, just open an issue in the new repo (or fusionauth-issues repo)

Reply:
Sounds good. I have a long list of things I think might be useful in the CLI, but would like to hear from the community about it first. Let's wait a bit.

Reply:
I have updated the pull request.

I have created a new repository for the example https://github.com/sonderformat-llc/fusionauth-example-tailwind-daisyui/
The repo would replace https://github.com/FusionAuth/fusionauth-example-theme-tailwind-daisyui

There was an issue in `@fusionauth/cli` that was fixed, could you please build and publish the new version?

I will prepare a pull request for the theme helper repository.

Reply:
@ColinFrick I updated fusionauth-cli